components/script/script_thread.rs

```diff
@@ -1635,7 +1635,13 @@ impl ScriptThread {
     fn handle_frame_load_event(&self, parent_id: PipelineId, frame_id: FrameId, child_id: PipelineId) {
         let iframe = self.documents.borrow().find_iframe(parent_id, frame_id);
         match iframe {
-            Some(iframe) => iframe.iframe_load_event_steps(child_id),
+            Some(iframe) => {
+                // If the iframe has no "src" attribute, a load event will have already been fired
+                // when binding the iframe to the tree.
+                if iframe.upcast::<Element>().has_attribute(&local_name!("src")) {
```

Comment:
Hmm, right, now I see that case. I couldn't reproduce it, though (I tried this:
```html
<html>
  <body>
  </body>
  <iframe id="iframe" onload="alert('onload');"></iframe>
  <script type="text/javascript">
    document.getElementById("iframe").src = "a.html";
  </script>
</html>
```
)

Does the spec say anything on what should happen to the first load event in the case of changing the `src`? I looked at your commit, but if I got it correctly it will only prevent the load steps from being loaded as long as the `IFrameLoadEventSteps` runnable checks the cancel flag after it is set, right? I'm imagining a case in which the program is exactly before `this.iframe_load_event_steps(self.pipeline_id);`, but after the `if` condition has been evaluated, and at this point the src changes and the other thread tries to cancel the event. The check makes it even harder for both events to run, but does it really guarantee it?
If you think it helps I can merge that commit into this PR. What do you think?